emukit/sensitivity/monte_carlo/monte_carlo_sensitivity.py

```diff
@@ -99,3 +99,22 @@ def compute_effects(self, main_sample: np.ndarray=None, fixing_sample: np.ndarra
 
             var_index += 1
         return main_effects, total_effects, total_variance
+
+
+
+class ModelBasedMonteCarloSensitivity(MonteCarloSensitivity):
+    '''
+    Class to compute the sensitivity coefficients of given model. This class wraps the model and calls the mean
+    predictions that are used to compute the sensitivity inputs using Monte Carlo.
+    '''
+
+    def __init__(self, model: IModel, input_domain: ParameterSpace)-> None:
+        '''
+        :param model: model wrapper with the interface IModel.
+        :param input_domain: space class.
+        '''
+
+        self.model = model
+        self.model_objective = lambda x : self.model.predict(x)[0].flatten()
```

Comment:
Would it be possible to get rid of using `flatten()` in this case? Generally using things like `flatten` or `atleast_2d` is dangerous, as it does not require people to be strict about the dimensions of the data. And that may lead to fucky behavior sometimes. So I'd rather avoid it, if possible

Reply:
It is actually not needed as the function is wrapped again.